src/interactable_spikes.c

```diff
@@ -0,0 +1,992 @@
+#include <string.h> // for memcpy
+
+#include "global.h"
+#include "task.h"
+#include "m4a.h"
+#include "malloc_vram.h"
+
+#include "interactable.h"
+
+#include "constants/animations.h"
+#include "constants/interactables.h"
+#include "constants/move_states.h"
+#include "constants/songs.h"
+
+typedef struct {
+    /* 0x00 */ SpriteBase base;
+    /* 0x0C */ Sprite s;
+    /* 0x3C */ u32 unk3C[2];
+} Sprite_Spikes; /* size: 0x44 */
+
+const u16 sSpikesOfZone[NUM_COURSE_ZONES + 1] = {
```

Comment:
btw I don't see a need in specifying array length here when the compiler can infer it

Reply:
"Future proofing" in case someone wants to create a game with more zones (which, granted, is unlikely, but better do it that way than the code reading garbage bytes in the end)